### ISSUE TITLE
CSS modes documentation is incorrect

### ISSUE THREAD

**Issue by mike-engel** [drwpow/cobalt-ui#72]
The documentation for `modeSelectors` in the css plugin has multiple examples, some of which are failing. On the [css plugin docs](https://cobalt-ui.pages.dev/docs/plugins/css/), modes do not include the `#` selector in the key, but on the [modes guide](https://cobalt-ui.pages.dev/docs/guides/modes/) they do. Looking at [this commit from Jan 2022](https://github.com/drwpow/cobalt-ui/commit/19eb0ac5c6b41f75ff0a99c0b1bea288c8a28007), the `#` is definitely required, so right now the css plugin docs are incorrect.

However, I'd really like to see the `#` requirement dropped, since we have modes defined at the semantic level rather than the color level (e.g. `background-strong.light`, `background-strong.dark` which alias to static colors). I'm not sure if you were planning on dropping the `#` requirement, or if there's a typo in the plugin docs.

**Comment by drwpow** [comment]
Yes docs definitely need to be updated in areas to remove the consistency.

Oh and you’re saying you aren’t using `$extensions.mode` at all? And you want to do something like:

```js
  modeSelectors: {
    '*.light': ['@prefers (color-scheme: light)'],
    '*.dark': ['@prefers (color-scheme: dark)']
  }
```

?

**Comment by mike-engel** [comment]
Right now I'm just doing some tests with modes, but I am using `$extensions.mode`:

```
{
	"background": {
		"$type": "color",
		"$description": "Background colors. Backgrounds should always be an alias to a color token.",
		"default": {
			"$value": "{colors.gray.200}",
			"$extensions": {
				"mode": {
					"light": "{colors.gray.200}",
					"dark": "{colors.gray.900}"
				}
			}
		},
		"hover": {
			"$value": "{colors.gray.300}"
		},
		"disabled": {
			"$value": "{colors.gray.100}"
		},
		"weak": {
			"default": {
				"$value": "{colors.gray.100}"
			},
			"hover": {
				"$value": "{colors.gray.200}"
			}
		},
		"strong": {
			"default": {
				"$value": "{colors.gray.300}"
			},
			"hover": {
				"$value": "{colors.gray.500}"
			}
		}
	},
}
```

But we have many root-level tokens that could use modes as we're doing something along the lines of asana's design token strategy: `background`, `selected`, `info`, `beta`, `warning`, etc. If we continued on with the `#` selector, we'd have a lot of duplication which is do-able, but annoying. If we add even more themes beyond simple light/dark (high contrast, tritanopia, etc.) then it gets even more annoying to manage:

```
modeSelectors: {
  'background#light': ['body[data-theme="light"]'],
  'background#dark': ['body[data-theme="dark"]'],
  'selected#light': ['body[data-theme="light"]'],
  'selected#dark': ['body[data-theme="dark"]'],
  'info#light': ['body[data-theme="light"]'],
  'info#dark': ['body[data-theme="dark"]'],
  // so on and so forth...
}
```

So, in summary, yeah, something like `*#light` or just `light` would be preferred

**Comment by drwpow** [comment]
Cool. Yeah might just need to add a simple [micromatch](https://github.com/micromatch/micromatch) support or something for this. I think that would be pretty trivial to do overall. The current restrictions aren’t intentional.

**Comment by drwpow** [comment]
Thought about it a little bit more, and would there be an opportunity to have a friendlier API here?

Something like:

```ts
  modeSelectors: [
    {
      mode: 'light',
      tokens: ['color.*', 'shadow.*'], // (optional) micromatch to filter to certain tokens
      css: ['@media (prefers-color-scheme: light)', 'body[data-color-mode="light"]'],
    },
    …
  ],
```

With the `#` symbol I was original thinking it would become shorthand for selecting modes. But over time I just think it’s confusing. And I’m not trying to write a token [DSL](https://developer.mozilla.org/en-US/docs/Glossary/DSL/Domain_specific_language) or anything 😄

**Comment by drwpow** [comment]
This would be a breaking change, but it might help usability? Open to thoughts.

**Comment by mike-engel** [comment]
I think that looks much better and straightforward compared to what's there now and what we talked about earlier. I think `selector` is a better name than `css` for the key, but that's a very small nitpick 😛 

This could also be done in a backwards compatible way, since you could easily transform existing `modeSelectors` into the new structure:

```js
modeSelectors: {
  'background#light': ['body[data-theme="light"]'],
}

// Gets turned into this internally
modeSelectors: [
  {
    mode: 'light',
    tokens: ['background.*'],
    css: ['body[data-theme="light"]'],
  },
]
```

I'm not sure how much you care about backwards compat, but it would be a simple loop through the keys of the old `modeSelectors` object to turn it into an array of objects

**Comment by drwpow** [comment]
Cool! Let’s go with this format, then. With the `selector` change you mentioned (that was just an oversight).

And as you pointed out, backwards compatibility is cheap here, so why not keep it and avoid a breaking change? But the docs can use the new format

**Comment by drwpow** [comment]
Just released `@cobalt-ui/plugin-css@1.4.0` which adds the API discussed. Please give it a try and let me know if you run into any weirdness 🙂

**Comment by mike-engel** [comment]
Thanks @drwpow! Finally got time for a quick test and everything looks great 🎉
